javascript/ql/src/Security/CWE-352/MissingCsrfMiddleware.ql

```diff
@@ -97,11 +97,72 @@ DataFlow::CallNode csrfMiddlewareCreation() {
   )
 }
 
+/**
+ * Gets a data flow node that flows to the base of a write to `cookies`, `session`, or `user`,
+ * where the written property has `csrf` or `xsrf` in its name.
+ */
+private DataFlow::SourceNode nodeLeadingToCsrfWrite(DataFlow::TypeBackTracker t) {
+  t.start() and
+  exists(DataFlow::PropRef value |
```

Comment:
Wouldn't this be simpler if we inline `value`? (Or if not, change its type to `PropWrite`)